langchain-core/src/vectorstores.ts

```diff
@@ -58,12 +58,13 @@ export type VectorStoreRetrieverInput<V extends VectorStoreInterface> =
     );
 
 export interface VectorStoreRetrieverInterface<
-  V extends VectorStoreInterface = VectorStoreInterface
+  V extends VectorStoreInterface<Metadata> = VectorStoreInterface<Metadata>,
+  Metadata extends Record<string, any> = Record<string, any>
 > extends BaseRetrieverInterface {
```

Comment:
I think the `Metadata` type should be passed to `BaseRetrieverInterface` too
```suggestion
  Metadata extends Record<string, any> = Record<string, any>
> extends BaseRetrieverInterface<Metadata> {
```

Reply:
I noticed this is also missing [here](https://github.com/langchain-ai/langchainjs/pull/4568/files#:~:text=%3E-,extends%20Runnable%3Cstring%2C%20DocumentInterface%3CMetadata%3E%5B%5D,implements%20BaseRetrieverInterface,-%7B)

If you could push up a commit updating that to, that would be great!